tests/executables_src/testDoubleBuffering.cpp

```diff
@@ -160,10 +219,263 @@ template<typename Register>
+    buf0 = backdoor->getRegisterAccessor<float>(c.firstBufferReg, 0, 0, {});
+    buf1 = backdoor->getRegisterAccessor<float>(c.secondBufferReg, 0, 0, {});
+  }
+
+  void simpleCheckExtractedChannels(std::string readerAReg) {
+    /*
+     * simple test for access to extracted channels of multiplexed 2D region
+     */
+    writingBufferNum->accessData(0) = 1;
+    writingBufferNum->write();
+
+    float modulation = 4.2; // example data
+    unsigned channel = 3;   // must match with xlmap
+    buf0->accessData(channel, 0) = modulation;
+    buf1->accessData(channel, 0) = 2 * modulation;
+    buf0->write();
+    buf1->write();
+
+    boost::barrier waitForBufferSwapStart{2}, waitForBufferSwapDone{2};
+    std::thread readerA([&] {
+      auto accessorA = d.getOneDRegisterAccessor<float>(readerAReg);
+      accessorA.readLatest();
+      // since writingBufferNum = 1, we expect buf0 contents to be read
+      BOOST_CHECK_CLOSE(accessorA[0], modulation, 1e-4);
+      waitForBufferSwapStart.wait();
+      waitForBufferSwapDone.wait();
+      accessorA.readLatest();
+      BOOST_CHECK_CLOSE(accessorA[0], 2 * modulation, 1e-4);
+    });
+
+    waitForBufferSwapStart.wait();
+    writingBufferNum->accessData(0) = 0;
+    writingBufferNum->write();
+    waitForBufferSwapDone.wait();
+
+    readerA.join();
+  }
+
+  void checkExtractedChannels(std::string readerAReg, std::string readerBReg) {
+    /*
+     * test access to extracted channels of multiplexed 2D region
+     * this is an application of concurrent readers
+     */
+
+    writingBufferNum->accessData(0) = 1;
+    writingBufferNum->write();
+
+    float modulation = 4.2;  // example data series 1
+    float correction = 10.1; // example data series 2
+    buf0->accessData(3, 0) = modulation;
+    buf1->accessData(3, 0) = 2 * modulation;
+    buf0->accessData(1, 0) = correction;
+    buf1->accessData(1, 0) = 2 * correction;
+    buf0->write();
+    buf1->write();
+
+    boost::barrier waitForBufferSwap{3};
+    std::thread readerA([&] {
+      auto accessorA = d.getOneDRegisterAccessor<float>(readerAReg);
+      accessorA.readLatest();
+      BOOST_CHECK_CLOSE(accessorA[0], modulation, 1e-4);
+      waitForBufferSwap.wait();
+      waitForBufferSwap.wait();
+      accessorA.readLatest();
+      BOOST_CHECK_CLOSE(accessorA[0], 2 * modulation, 1e-4);
+    });
+    std::thread readerB([&] {
+      auto accessorB = d.getOneDRegisterAccessor<float>(readerBReg);
+      accessorB.read();
+      BOOST_CHECK_CLOSE(accessorB[0], correction, 1e-4);
+      waitForBufferSwap.wait();
+      waitForBufferSwap.wait();
+      accessorB.read();
+      BOOST_CHECK_CLOSE(accessorB[0], 2 * correction, 1e-4);
+    });
+
+    waitForBufferSwap.wait();
+    writingBufferNum->accessData(0) = 0;
+    writingBufferNum->write();
+    waitForBufferSwap.wait();
+
+    readerA.join();
+    readerB.join();
+
+    // Check that also reading from a TransferGroup works
```

Comment:
What does this test? The test is completely linear. It would have worked without double buffering.

Reply:
The test was failing before TransferGroup support was right